server/routes/applications/applicationDetailsRoutes.ts

```diff
@@ -24,10 +25,17 @@ export default function applicationDetailsRoutes({ auditService }: { auditServic
         return res.redirect(URLS.APPLICATION_TYPE)
       }
 
+      const additionalData = applicationData?.additionalData || {}
+      const details = (additionalData as SwapVOsForPinCreditDetails).details || ''
```

Comment:
Should `additionalData` not either have it's own type or be a combination of the others? Because the additional data is going to be `SwapVOsForPinCreditDetails` or `AddEmergencyPinPhoneCreditDetails` and not both in the same request so this isn't using the types properly, rather it's just fixing the validation error.